Makefile

```diff
@@ -84,16 +85,12 @@ vis-single: vis-single.c vis-single-payload.inc
 	${STRIP} $@
 
 docker: clean
-	docker build -t vis .
-	docker run --rm -d --name vis vis tail -f /dev/null
-	docker exec vis apk update
-	docker exec vis apk upgrade
-	docker cp . vis:/tmp/vis
-	docker exec vis sed -i '/^VERSION/c VERSION = $(VERSION)' Makefile
-	docker exec vis ./configure CC='cc --static' --enable-acl
-	docker exec vis make clean vis-single
-	docker cp vis:/tmp/vis/vis-single vis
-	docker kill vis
+	$(DOCKER) build -t vis .
+	$(DOCKER) run --rm --name vis -v $(PWD):/build vis ./configure CC='cc --static' --enable-acl
+	$(DOCKER) run --rm --name vis -v $(PWD):/build vis make VERSION="$(VERSION)" clean vis-single
```

Comment:
This sets up a bind mount which makes the current working directory of the host available to the guest. I would prefer if we kept the more rigorous isolation and explicitly copy what is needed. To copy the source tree we could probably add an appropriate `COPY` instruction to  the Dockerfile. For the reverse direction `docker cp` is probably needed.

Reply:
The idea of docker is to build an image that has everything you need from a `Dockerfile`. The work is done running short-living and volatile containers, persistent data is stored in volumes.

Our current code starts a background container and executes commands inside this running container. If the build process breaks for whatever reason you have left over containers - running or stopped. IMHO it is really annoying to clean up a stopped or broken docker build process.

Reply:
Containers can more easily be cleaned up than my working directory. I don't want to expose it to the container and let it leave files (with wrong UIDs etc) behind.